src/reflect/scala/reflect/internal/util/Statistics.scala

```diff
@@ -267,19 +308,20 @@ quant)
    */
   final val hotEnabled = false
 
-  def enabled = _enabled
+  @inline def enabled = canEnable
   def enabled_=(cond: Boolean) = {
-    if (cond && !_enabled) {
+    if (cond && !canEnable) {
       val start = System.nanoTime()
       var total = 0L
       for (i <- 1 to 10000) {
         val time = System.nanoTime()
         total += System.nanoTime() - time
       }
       val total2 = System.nanoTime() - start
+      // Shouldn't we replace this println with a `reporter.info`?
       println("Enabling statistics, measuring overhead = "+
               total/10000.0+"ns to "+total2/10000.0+"ns per timer")
-      _enabled = true
+      this.setValue(true)
```

Comment:
Note that here we only can enable, but we cannot disable. Is this what we want, or would we like to enable/disable them for hot compiler? Personally, I think we could support both. It's true it will have an impact and will force the deoptimization of bunch a code, but this is the case when we enable statistics anyway.